lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1416,13 +1416,32 @@ synthesizeObservedSetterBody(AccessorDecl *Set, TargetImpl target,
   SmallVector<ASTNode, 6> SetterBody;
 
   auto callObserver = [&](AccessorDecl *observer, VarDecl *arg) {
+    // If we're not loading the old value, then re-write the observer's type to
+    // be either () -> () or (Self) -> () -> () and remove the oldValue
+    // parameter.
+    if (!arg) {
```

Comment:
Can you find a way to decide to add the oldValue parameter or not based on whether it’s used instead of removing it later?